### PR TITLE
Made PeerActor route messages without contacting PeerManager

### PR DESCRIPTION
**Pull request by pompon0**
this required moving a few methods of PeerManager to NetworkState, but that was the long term plan anyway.

### CODE REVIEW

File: chain/network/src/peer/peer_actor.rs [hunk 1]
```diff
@@ -418,7 +425,7 @@ impl PeerActor {
             if let Some(peer_id) = self.other_peer_id() { peer_id.clone() } else { return };
 
         // Wrap peer message into what client expects.
-        let network_client_msg = match msg {
+        let network_client_msg = match msg.clone() {
```
Comment (mm-near) [inline chain/network/src/peer/peer_actor.rs:428]: we're doing a bunch of "clone" calls -- let's make sure that we don't duplicate something expensive (like large incoming network message)

**Review by mm-near — APPROVED** [review]

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 2]
```diff
@@ -215,6 +215,86 @@ impl NetworkState {
         });
         PartialEdgeInfo::new(&self.config.node_id(), peer1, nonce, &self.config.node_key)
     }
+
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/peer_manager/peer_manager_actor.rs:218]: yes, already scheduled

File: chain/network/src/peer/peer_actor.rs [hunk 3]
```diff
@@ -1078,30 +1087,61 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
...
 
                 // Receive invalid routed message from peer.
-                if !routed_message.verify() {
+                if !msg.verify() {
                     self.ban_peer(ctx, ReasonForBan::InvalidSignature);
+                    return;
+                }
+                let from = self.other_peer_id().unwrap().clone();
+                if msg.expect_response() {
+                    tracing::trace!(target: "network", route_back = ?msg.clone(), "Received peer message that requires");
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/peer/peer_actor.rs:1104]: done

File: chain/network/src/peer/peer_actor.rs [hunk 4]
```diff
@@ -1078,30 +1087,61 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
...
-                            }
-                            actix::fut::ready(())
-                        })
-                        .spawn(ctx);
+                    let mut msg = (*msg).clone();
+                    if msg.decrease_ttl() {
+                        self.network_state.send_signed_message_to_peer(&self.clock, Arc::new(msg));
+                    } else {
+                        self.network_state.config.event_sink.push(Event::RoutedMessageDropped);
+                        warn!(target: "network", ?msg, ?from, "Message dropped because TTL reached 0.");
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/peer/peer_actor.rs:1143]: done

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 5]
```diff
@@ -215,6 +215,86 @@ impl NetworkState {
...
+        &self,
+        clock: &time::Clock,
+        msg: Arc<RoutedMessageV2>,
+    ) -> bool {
+        let my_peer_id = self.config.node_id();
+
+        // Check if the message is for myself and don't try to send it in that case.
+        if let PeerIdOrHash::PeerId(target) = &msg.msg.target {
+            if target == &my_peer_id {
+                debug!(target: "network", account_id = ?self.config.validator.as_ref().map(|v|v.account_id()), ?my_peer_id, ?msg, "Drop signed message to myself");
```
[the remark this answers is not in the document]
Comment (pompon0) [reply chain/network/src/peer_manager/peer_manager_actor.rs:268]: done

**Comment by matklad** [comment]
Do we want to nayduck this? I am noticing that network refactors tend to break nayduck sometimes

**Pull request merged**